Gems/PhysX/Code/Tests/PhysXColliderComponentModeTests.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "TestColliderComponent.h"
+
+#include <AZTestShared/Math/MathTestHelpers.h>
+#include <AzManipulatorTestFramework/AzManipulatorTestFrameworkTestHelpers.h>
+#include <AzManipulatorTestFramework/IndirectManipulatorViewportInteraction.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/EditorNonUniformScaleComponent.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Viewport/ViewportSettings.h>
+#include <AzToolsFramework/ViewportSelection/EditorDefaultSelection.h>
+#include <AzToolsFramework/ViewportSelection/EditorInteractionSystemViewportSelectionRequestBus.h>
+#include <AzToolsFramework/ViewportSelection/EditorSelectionUtil.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+#include <EditorColliderComponent.h>
+#include <Tests/Viewport/ViewportUiManagerTests.h>
+
+namespace UnitTest
+{
+    class PhysXColliderComponentModeTest : public ToolsApplicationFixture<false>
+    {
+    protected:
+        using EntityPtr = AZ::Entity*;
+
+        AZ::ComponentId m_colliderComponentId;
+
+        EntityPtr CreateColliderComponent()
+        {
+            AZ::Entity* entity = nullptr;
+            AZ::EntityId entityId = CreateDefaultEditorEntity("ComponentModeEntity", &entity);
+
+            entity->Deactivate();
+
+            // Add placeholder component which implements component mode.
+            auto colliderComponent = entity->CreateComponent<TestColliderComponentMode>();
+
+            m_colliderComponentId = colliderComponent->GetId();
+
+            entity->Activate();
+
+            AzToolsFramework::SelectEntity(entityId);
+
+            return entity;
+        }
+
+        // Needed to support ViewportUi request calls.
+        ViewportManagerWrapper m_viewportManagerWrapper;
+
+        void SetUpEditorFixtureImpl() override
+        {
+            m_viewportManagerWrapper.Create();
+        }
+        void TearDownEditorFixtureImpl() override
+        {
+            m_viewportManagerWrapper.Destroy();
+        }
```

Comment:
Please move implementation to `PhysXColliderComponentModeTests.cpp`

Reply:
what would be the best way to include this file as a cpp?

Reply:
Just move the implementation -  I would argue these smaller functions are okay to leave in the grand scheme of things, but maybe the larger ones, leave the interface here, and then put the body in the .cpp

```
// .h

...
void SetUpEditorFixtureImpl() override;

// .cpp

void <typename>::SetUpEditorFixtureImpl()
{
    // implementation
}
```